ghost/core/core/frontend/web/middleware/frontend-caching.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * @file Middleware to set the appropriate cache headers on the frontend
+ */
+const config = require('../../../shared/config');
+const shared = require('../../../server/web/shared');
+const {api} = require('../../services/proxy');
+
+/**
+ * Calculate the member's active tier.
+ * @param {object} member - The member object.
+ * @param {object} freeTier - The free tier object.
+ * @returns {string|null} - The member's active tier, or null if the member has more than one active subscription.
+ */
+function calculateMemberTier(member, freeTier) {
+    const activeSubscriptions = member.subscriptions.filter(sub => sub.status === 'active');
+    if (activeSubscriptions.length === 0) {
+        return freeTier; // how do we get the free tier's ID in this context? 
+    } 
+    if (activeSubscriptions.length === 1) {
+        return activeSubscriptions[0].tier;
+    }
+    return null; // More than one active subscription
+}
+
+/**
+ * @typedef {function(): Promise<object>} GetFreeTier
+ */
+
+/**
+ * Returns the frontend caching middleware.
+ * @param {GetFreeTier} [getFreeTier] - Async function that takes no arguments and resolves to the free tier object.
+ * @returns {function} Middleware function.
+ */
+const getMiddleware = async (getFreeTier = async () => {
```

Comment:
We need a way to get the ID of the free tier in the middleware, but for the sake of unit testing I've tried to do a bit of DI here. It works, and it makes testing easier, but I wonder if there is a better way to do this that I've overlooked. Feels a bit overly complicated for what it's doing.